pkg/clusterregistryconfig/flags.go

```diff
@@ -0,0 +1,252 @@
+		nil,
+		"A comma-separated list of registries for which image pull and push actions are denied.",
+	)
+
+	cmd.Flags().StringVar(
+		&args.allowedRegistriesForImport,
+		allowedRegistriesForImportFlag,
+		"",
+		"Limits the container image registries from which normal users can import images. "+
+			"The format should be a comma-separated list of 'domainName:insecure'. "+
+			"'domainName' specifies a domain name for the registry. "+
+			"'insecure' indicates whether the registry is secure or insecure.",
+	)
+
+	cmd.Flags().StringVar(
+		&args.platformAllowlist,
+		platformAllowlistFlag,
+		"",
+		"The id of the internal registry that needs to be whitelisted for the platform to work. "+
+			"It can be omitted at creation and updating and its lifecycle can be managed separately if needed.",
+	)
+	cmd.Flags().MarkHidden("platform-allowlist")
+
+	cmd.Flags().StringVar(
+		&args.additionalTrustedCa,
+		additionalTrustedCaFlag,
+		"",
+		"A json file containing the registry hostname as the key,"+
+			" and the PEM-encoded certificate as the value, for each additional registry CA to trust.")
+
+	return args
+}
+
+func GetClusterRegistryConfigOptions(cmd *pflag.FlagSet, args *ClusterRegistryConfigArgs) (
+	*ClusterRegistryConfigArgs, error) {
+	var err error
+
+	result := &ClusterRegistryConfigArgs{}
+
+	result.allowedRegistries = args.allowedRegistries
+	result.insecureRegistries = args.insecureRegistries
+	result.blockedRegistries = args.blockedRegistries
+	result.additionalTrustedCa = args.additionalTrustedCa
+	result.allowedRegistriesForImport = args.allowedRegistriesForImport
+	result.platformAllowlist = args.platformAllowlist
+
+	if !IsClusterRegistryConfigSetViaCLI(cmd) {
+		if !interactive.Enabled() {
+			return nil, nil
+		}
+	}
+
+	// if interactive.Enabled() && !cmd.Changed(usernameFlag) {
+	// 	result.username, err = interactive.GetString(interactive.Input{
+	// 		Question: "Username",
+	// 		Default:  result.username,
+	// 		Help:     cmd.Lookup(usernameFlag).Usage,
+	// 	})
+	// 	if err != nil {
+	// 		return nil, err
+	// 	}
+	// }
+
+	// if interactive.Enabled() && !cmd.Changed(expirationFlag) {
+	// 	inputString, err := interactive.GetString(interactive.Input{
+	// 		Question: "Expiration duration",
+	// 		Default:  result.expirationDuration,
+	// 		Help:     cmd.Lookup(expirationFlag).Usage,
+	// 	})
+	// 	if err != nil {
+	// 		return nil, err
+	// 	}
+	// 	if inputString != "" {
+	// 		duration, err := time.ParseDuration(inputString)
+	// 		if err != nil {
+	// 			return nil, err
+	// 		}
+	// 		result.expirationDuration = duration
+	// 	}
+	// }
+
+	return result, err
+}
+
+func IsClusterRegistryConfigSetViaCLI(cmd *pflag.FlagSet) bool {
+	for _, parameter := range []string{allowedRegistriesFlag,
+		insecureRegistriesFlag, blockedRegistriesFlag} {
+
+		if cmd.Changed(parameter) {
+			return true
+		}
+	}
+
+	return false
+}
+
+func BuildRegistryConfig(args *ClusterRegistryConfigArgs) (*cmv1.ClusterRegistryConfigBuilder, error) {
```

Comment:
introduced the panic issue `panic: runtime error: invalid memory address or nil pointer dereference`
Looking...